python/geoarrow/dataset.py

```diff
@@ -0,0 +1,139 @@
+from concurrent.futures import ThreadPoolExecutor, wait
+
+import pyarrow as _pa
+import pyarrow.dataset as _ds
+import pyarrow.compute as _compute
+from . import pyarrow as _ga
+from .pyarrow._kernel import Kernel
+
+
+def _index_fragment(fragment, column):
```

Comment:
nit: I personally prefer having internal functions be part of the class (assuming they don't have a purpose outside of the class).